### PR TITLE
[#213] Remove set_quorum_threshold and its limits from config

### LINKED ISSUE #213: Remove set_quorum_threshold
# Clarification and motivation

The motivation for this issue is specular to #212 but for the quorum threshold.

Here too we also won't need the limits anymore and we can remove them from the contract `storage`.

# Acceptance criteria

The `set_quorum_threshold` entrypoint is no longer part of the contract.

### PR DESCRIPTION
**Pull request by sras**
## Description

Remove the 'set_quorum_threshold' entrypoint and the associated code and the
max_quorum_threshold, min_quorum_threshold limits that are part of the
configuration in storage.

## Related issue(s)

Resolves #213 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: docs/specification.md [hunk 1]
```diff
@@ -689,33 +684,6 @@ Parameter (in Michelson):
...
-```
-(pair %set_quorum_threshold (nat %numerator) (nat %denominator))
-```
-
-- Update the quorum threshold which proposals have to meet to not get rejected.
-- The Quorum threshold is calculated as the proportion of total votes (upvotes
-  and downvotes) over the frozen token's total supply.
-- This affects all ongoing and new proposals.
-- Fails with `NOT_ADMIN` if the sender is not the administrator.
-- Fails with `OUT_OF_BOUND_QUORUM_THRESHOLD` if the quorum threshold value is out of the bound set by the configuration or the numerator is not smaller than the denominator.
```
Comment (pasqu4le) [inline docs/specification.md:717]: Not that this entrypoint has been removed the `OUT_OF_BOUND_QUORUM_THRESHOLD` is no longer used.

We should remove it from the `Errors` section in this doc as well as removing the instances in `Ligo.BaseDAO.Types`

File: src/proposal.mligo [hunk 2]
```diff
@@ -285,17 +285,6 @@ let set_voting_period(new_period, config, store : voting_period * config * stora
   let store = { store with voting_period = new_period; last_period_change = vp_log } in
   (([] : operation list), store)
 
-// Update quorum_threshold. The new quorum_threshold affects
-// all ongoing and new proposals.
-[@inline]
-let set_quorum_threshold(new_threshold, config, store : quorum_threshold * config * storage): return =
-  let store = authorize_admin store in
-  if   is_le_qt (new_threshold, config.max_quorum_threshold)
```
Comment (pasqu4le) [inline src/proposal.mligo:293]: The definition of `is_le_qt` can now also be removed, as this was its only use.

**Review by pasqu4le — COMMENTED** [review]
Only two small points.
Ah and the commits need to be fixed as well.

File: docs/specification.md (hunk 1 above)
Comment (sras) [reply docs/specification.md:658]: Updated

File: src/proposal.mligo (hunk 2 above)
Comment (sras) [reply src/proposal.mligo:293]: Updated

**Review by pasqu4le — APPROVED** [review]

**Pull request merged**